src/memory-management.md

```diff
@@ -13,3 +13,9 @@ Rust offers a new mix:
 It does this with an explicit ownership concept.
 
 First, let's refresh how memory management works.
+
+<details>
+  
+One way to think about it, is that the compiler correctly puts all `alloc` and `free` calls in the fixed locations for you, without resorting to the garbage collection.
```

Comment:
It's not the compiler doing that though, it's whatever heap allocated types you use, like `Box` or `Arc`.

Reply:
I think what you said applies also to C++; idiomatic C++ seldom requires `free`. So, it is not a unique feature of Rust.

`memory_management/rust.md` clearly explains the differences from other languages. I don't think we need to add any detail at this page.

Reply:
> "One way to think about it, is that Rust correctly computes where to allocate and free memory for you, as if you did it by hand and never made a mistake".

This is also how I think of it. 

> I think what you said applies also to C++; idiomatic C++ seldom requires `free`. So, it is not a unique feature of Rust.

I agree :slightly_smiling_face:

Reply:
@qwandor, please work with @igor-petruk to land this — or close it if you agree that we don't need it.

Reply:
I think we should just leave this out, it's incorrect and misleading. The compiler doesn't insert any alloc or free calls.

Reply:
Hi @qwandor, we should close this now that #1255 is merged, right?

Reply:
Happy to revert if it is more misleading than helpful.